langlib/lang.__internal/src/main/java/org/ballerinalang/langlib/internal/GetElementType.java

```diff
@@ -36,22 +34,23 @@
  */
 public class GetElementType {
 
+    private GetElementType() {
+    }
+
     public static BTypedesc getElementType(Object td) {
         BTypedesc bTypedesc = (BTypedesc) td;
-        return getTypeDescValue(bTypedesc.getDescribingType());
+        return getElementTypeDescValue(bTypedesc.getDescribingType());
     }
 
-    private static BTypedesc getTypeDescValue(Type type) {
-        if (type.getTag() == TypeTags.ARRAY_TAG) {
-            return ValueCreator.createTypedescValue(((ArrayType) type).getElementType());
-        } else if (type.getTag() == TypeTags.STREAM_TAG) {
-            return ValueCreator.createTypedescValue(((StreamType) type).getConstrainedType());
-        } else if (type.getTag() == TypeTags.TABLE_TAG) {
-            return ValueCreator.createTypedescValue(((TableType) type).getConstrainedType());
-        } else if (type.getTag() == TypeTags.FINITE_TYPE_TAG && ((FiniteType) type).getValueSpace().size() == 1) {
-            return getTypeDescValue(((BValue) (((FiniteType) type).getValueSpace().iterator().next())).getType());
+    private static BTypedesc getElementTypeDescValue(Type type) {
```

Comment:
Is the immutable case already covered in the tests? If not let's add a test

Reply:
It seems it is already covered in the test you have written.
https://github.com/Nadeeshan96/ballerina-lang/blob/0446326d33708a48f644fc1b2a5385028f29074f/langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java#L576 
https://github.com/Nadeeshan96/ballerina-lang/blob/0446326d33708a48f644fc1b2a5385028f29074f/langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal#L1296

Reply:
Let's add one for the tuple case

Reply:
I will do it.